src/client/components/map/MapToken.tsx

```diff
@@ -262,6 +267,39 @@ function MapTokenInner({
       )}
     </Container>
   );
+  const notifications = useServerState((state) => state.logEntries);
+  const [lastRolled, setLastRolled] = useState<RRLogEntryDiceRoll | null>();
+  useEffect(() => {
+    const list = entries(notifications);
+
+    function isNotTooOld(entry: RRLogEntry) {
+      return Date.now() - entry.timestamp < DICE_ROLL_DISPLAY_DURATION;
+    }
+
+    function hasRightCharacter(entry: RRLogEntryDiceRoll) {
+      return entry.payload.characterIds?.includes(character.id);
+    }
+
+    let selectedEntry: RRLogEntry | null = null;
+    for (
+      let i = list.length - 1, entry = list[i];
+      entry && !selectedEntry && isNotTooOld(entry);
+      entry = list[--i]
+    )
+      if (entry.type === "diceRoll" && hasRightCharacter(entry)) {
+        selectedEntry = entry;
+      }
```

Comment:
I find the lack of brackets on the for loop a little sketchy :D

Reply:
Fair enough :D